exasol_advanced_analytics_framework/udf_communication/socket_factory/abstract_socket_factory.py

```diff
@@ -0,0 +1,99 @@
+import abc
+from enum import Enum, auto
+from typing import Union, List, Set, Optional
+
+
+class Frame(abc.ABC):
+    @abc.abstractmethod
+    def to_bytes(self) -> bytes:
+        pass
```

Comment:
why not require `__bytes__` to be implemented?

```python
id_frame = IdFrame(id=1, data="some data")

raw_data = bytes(id_frame)
socket.send(raw_data)

```

Reply:
The ZMQ-Frame is a bit special, it is actually a placeholder which keeps a memory buffer in C alive, and my abstraction follows this.

Reply:
you still could keep the buffer alive, but anyway was just an idea.